### ISSUE TITLE
user profile screens need a flag / report button to report bad profiles or display names

### ISSUE THREAD

**Issue by Alys** [HabitRPG/habitica#9486]
The profile screen for a player needs to have a report button, similar to the report button on chat messages, so that players can report profiles containing bad text or images, or players who have bad names.

See the comments below for details.
~This needs design work to determine the location and appearance of the report button (although I suggest it looks the same as it does on chat messages to help players recognise it). @Tressley~

~Once the design is done, I'll add some more details to this issue about how the flagging should work and what effects it should have.~

**Comment by Alys** [comment]
Labels updated (not yet sectioned, on hold needs design).

**Comment by Tressley** [comment]
This design lives in Zeplin: https://zpl.io/anxLEv2

**Comment by Alys** [comment]
~I'm putting this back on hold until I've added the details that I mentioned in the top post ("Once the design is done, I'll add some more details to this issue about how the flagging should work and what effects it should have.")~

**Comment by Alys** [comment]
The way this should work is:

**Report Modal**

When you click on the report option from the profile screen, a report modal should open, very similar to the one you see when you click report on a chat post (first screenshot below).

The differences between the chat post modal and the profile modal should be:
* Instead of seeing the text of a post ("Test post" in the screenshot below) you see "You are reporting the profile of @Username / Display Name", where "Username" and "Display Name" are the user's names of course, and they should be inserted into the locales string as a variable (rather than being hard-coded to appear at the end of the string).
* For all the text under that, each instance of "post" should be replaced with "profile". Create new strings for all of those (instead of modifying the original strings to use a variable for post versus profile). Give the string similar keys to the existing strings and put them in the English locales file immediately under the original string so that it's easy for translators to copy most of the translation from one string to another.
* Also after the word "violence" add ", inappropriate image, inappropriate name".
* The "Reset Flag Count" button should not appear. _(NB We still need it on the modal for reporting chat posts but only moderators and staff see it there. It's used to un-hide a chat post that's been hidden by reporting it.)_

![image](https://user-images.githubusercontent.com/1495809/87371284-58131c80-c5c8-11ea-8065-fd0b71f6ec53.png)

When the player clicks the "Report" button on the modal, an email and a Slack notification should be sent. The recipient/destination details for both should be the same as when a post is reported. More details about the email and Slack notification are below.

There should be no attempt to hide any parts of the profile or to mark it has having been reported. _(This is because profile violations are relatively rare and not highly visible, and we expect a fair proportion of false flags, and real violations will be handled by moderators fairly quickly. Hiding the profile is probably more trouble than it's worth.)_



**Email**

We'll need to create a new Mandrill template for the email but I or Habitica staff can do that - don't worry about the details of that just set up the code to send the email similar to how it's done for reporting a post.

The email should contain:
* The details of the person reporting the profile (just like the details of a person reporting a post).
* The optional comments that the person wrote in the report screen, if any (as for reporting a post). 
* The details of the person who owns the profile: Username, Display Name, email address, User ID (just like for reporting a post).
* The profile URL (e.g., https://habitica.com/profile/d904bd62-da08-416b-a816-ba797c9ee265 where the text after `profile/` is the User ID). Don't try to make this an "a href" link - just the raw URL is fine.
* The image from the profile (`profile.imageUrl`). If no image is specified, the email should say "no profile image". Don't try to make this an HTML image or "a href" tag, just the raw `profile.imageUrl` string is fine.
* The "About" text from the profile (`profile.blurb`). If that text doesn't exist, the email should say "no About text".

**Slack Notification**

The Slack notification should be similar to the one for reporting a post (screenshot below). It should contain:
* The details of the person reporting the profile (just like the details of a person reporting a post).
* The optional comments that the person wrote in the report screen, if any. This should be displayed as it is when you're reporting a post but if you could add a line break just before it, that would be good! I.e., if the screenshot below was for reporting a profile, there'd be a line break between "... and commented:" and "This is an optional comment." (NB: We want this line break for reporting posts as well so if it's convenient to add it for that too as you're fixing this issue, please do, but don't feel that you have to.)
* The details of the person who owns the profile (just like for reporting a post).
* The `preferences.language` value of the person reporting the profile. This can be formatted as "; language: fr" after the User ID, as it is for the reporter. (NB: We want this for reporting posts as well so if it's convenient to add it for that too as you're fixing this issue, please do, but don't feel that you have to.)
* The profile URL as a clickable link - that can go in the place where you'd see the link to the Guild name if you were reporting a post ("Flag in My Great Guild" in the screenshot below). The visible text for the link should be "Profile URL".
* The image from the profile (`profile.imageUrl`) as a clickable link with the visible text being the URL of the image. If no image is specified, the notification should say "no profile image". Do not try to make this an inline image - I think we'd prefer that we did not see the image itself in the notification (Slack might display the image but if so we'll put up with it).
* The "About" text from the profile (`profile.blurb`). If that text doesn't exist, the notification should say "no About text".

Note that the strings "no About text" and "no profile image" and "Profile URL" do not need to be translatable and should not be placed in json files under website/common/locales/en.

![image](https://user-images.githubusercontent.com/1495809/87619730-38166100-c760-11ea-9874-bfc1c3a45278.png)

If anyone has any questions or concerns about this, post here!

**Comment by RussWilkie** [comment]
Hello! I was just panning through the issues on here with an intent of taking up one. I'm noticing this issue is 7-years dated, but still marked as Open. The requested changes seem to differ slightly from the last comment, but I do notice the report a player option does generally meet the bullet points. I was wondering if this was still something I could(or should) work on.

**Comment by Tressley** [comment]
@RussWilkie -- Thanks for the ping! This issue looks to have slipped through the cracks and has now been closed.
